tools/shared.py

```diff
@@ -1204,6 +1208,8 @@ def __setattr__(self, attr, value):
           exit_with_error('Invalid command line option -s ' + attr + '=' + str(value) + ': ' + error_message)
         else:
           logger.debug('Option -s ' + attr + '=' + str(value) + ' has been removed from the codebase. (' + error_message + ')')
+        if attr in SettingsManager.renamed_settings:
```

Comment:
This depends on all renamed settings being a subset of legacy settings, which is handled in settings.js. The data here has an unmodeled dependency, which makes me a bit uneasy.